ui/modules/things/thingMessages.ts

```diff
@@ -0,0 +1,159 @@
+        dom.favIconThingMessage.classList.contains('bi-star-fill')) {
+      dom.favIconThingMessage.classList.replace('bi-star-fill', 'bi-star');
+      delete Environments.current().messageTemplates['/'][templateName];
+    } else {
+      dom.favIconThingMessage.classList.replace('bi-star', 'bi-star-fill');
+      Environments.current().messageTemplates['/'][templateName] = {
+        subject: dom.inputThingMessageSubject.value,
+        timeout: dom.inputThingMessageTimeout.value,
+        payload: JSON.parse(acePayload.getValue()),
+      };
+      acePayload.session.getUndoManager().markClean();
+    }
+    Environments.environmentsJsonChanged('messageTemplates');
+  };
+
+  dom.ulThingMessageTemplates.addEventListener('click', (event) => {
+    if (event.target && event.target.classList.contains('dropdown-item')) {
+      dom.favIconThingMessage.classList.replace('bi-star', 'bi-star-fill');
+      const template = Environments.current().messageTemplates['/'][event.target.textContent];
+      dom.inputThingMessageTemplate.value = event.target.textContent;
+      dom.inputThingMessageSubject.value = template.subject;
+      dom.inputThingMessageTimeout.value = template.timeout;
+      acePayload.setValue(JSON.stringify(template.payload, null, 2), -1);
+      acePayload.session.getUndoManager().markClean();
+    }
+  });
+
+  [dom.inputThingMessageTemplate, dom.inputThingMessageSubject, dom.inputThingMessageTimeout].forEach((e) => {
+    e.addEventListener('change', () => {
+      dom.favIconThingMessage.classList.replace('bi-star-fill', 'bi-star');
+    });
+  });
+
+  acePayload.on('input', () => {
+    if (!acePayload.session.getUndoManager().isClean()) {
+      dom.favIconThingMessage.classList.replace('bi-star-fill', 'bi-star');
+    }
+  });
+}
+
+/**
+ * Calls Ditto to send a message with the parameters of the fields in the UI
+ */
+function messageThing() {
+  const payload = acePayload && acePayload.getValue().length > 0 && JSON.parse(acePayload.getValue());
+  aceResponse.setValue('');
+  API.callDittoREST('POST', '/things/' + Things.theThing.thingId +
+      '/inbox/messages/' + dom.inputThingMessageSubject.value +
+      '?timeout=' + dom.inputThingMessageTimeout.value,
+  payload,
+  ).then((data) => {
+    dom.buttonThingMessageSend.classList.remove('busy');
+    dom.buttonThingMessageSend.disabled = false;
+    if (dom.inputThingMessageTimeout.value > 0) {
+      aceResponse.setValue(JSON.stringify(data, null, 2), -1);
+    }
+  }).catch((err) => {
+    dom.buttonThingMessageSend.classList.remove('busy');
+    dom.buttonThingMessageSend.disabled = false;
+    aceResponse.setValue('');
+  });
+}
+
+function onEnvironmentChanged(modifiedField) {
+  Environments.current()['messageTemplates'] = Environments.current()['messageTemplates'] || {};
+
+  if (!modifiedField) {
+    clearAllFields();
+  }
+  if (modifiedField === 'messageTemplates') {
+    refillTemplates();
```

Comment:
there is no event to fill the template list initially. This can be fixed to remove the check for the modified field and call refillTemplates just on any environment change.

Reply:
@thfries do you also have a solution and can add it too a patch?

Reply:
Yes that is included in the patch. Let me know, if the patch file works.

Reply:
Yes, worked - thanks .. I added the patch as commit 👍